drivers/soundwire/stream.c

```diff
@@ -1388,22 +1406,23 @@ int sdw_stream_add_slave(struct sdw_slave *slave,
 	 * check if Master is already allocated, if so skip allocation
 	 * and go to configuration
 	 */
-	m_rt = sdw_find_master_rt(slave->bus, stream);
+	m_rt = sdw_master_rt_find(slave->bus, stream);
 	if (m_rt)
 		goto skip_alloc_master_rt;
 
 	/*
 	 * If this API is invoked by Slave first then m_rt is not valid.
 	 * So, allocate m_rt and add Slave to it.
 	 */
-	m_rt = sdw_alloc_master_rt(slave->bus, stream_config, stream);
+	m_rt = sdw_master_rt_alloc(slave->bus, stream);
 	if (!m_rt) {
 		dev_err(&slave->dev,
 			"alloc master runtime failed for stream:%s\n",
 			stream->name);
 		ret = -ENOMEM;
 		goto error;
 	}
+	ret =  sdw_master_rt_config(m_rt, stream_config);
```

Comment:
Although sdw_master_rt_config() always return 0, but to be consistent, should we add
```
if (ret < 0)
        goto error;
```
?

Reply:
yes, this is a miss, thanks @bardliao

Reply:
But this is removed in follow-up patches, so the update in the end looks exactly the same. See new branch